api/v1alpha1/nodepool_types.go

```diff
@@ -47,6 +51,13 @@ type NodePoolStatus struct {
 	// +optional
 	NodeCount  int                `json:"nodeCount"`
 	Conditions []metav1.Condition `json:"conditions"`
+
+	// Version is the semantic version of the release applied by
+	// the hosted control plane operator.
+	// For a nodePool a given version dictates the ignition config and
+	// an image artifact e.g an AMI in AWS.
+	// +kubebuilder:validation:Optional
+	Version string `json:"version,omitempty"`
```

Comment:
Do we need to have a `desiredVersion` in the spec and an `{actual}Version` in the status?  Setting something in the status to trigger reconciliation seems weird.  Or maybe I'm misunderstanding this.

Reply:
Same question. Even if the desired version field is owned/writable only by the hypershift operator, I expected the field to be present to serve as the basis of reconciliation. Otherwise you have to reverse engineer the desired state through inference based on the HostedCluster? Seems weird.

Reply:
The intention was to for the field to reflect current version of the nodePool. Expected version would be always the current cluster version given by `hostedCluster.Status.Version`

Reply:
If the version should propagate down from hostedcluster I'd expect the hostedcluster controller to update `nodepool.spec.version` and the nodepool controller to reconcile nodepools based on `nodepool.spec.version`. Otherwise the node pool controller is actually reconciling hostedclusters as the authoritative desired state instead of nodepools, which seems like a layering violation?